src/Loads.js

```diff
@@ -87,26 +92,35 @@ export default class Loads extends Component<Props, State> {
     this._setTimeouts();
     return fn(...args)
       .then(response => {
-        this.handleResponse({ response, event: 'SUCCESS' });
+        this.handleResponse({ response, event: EVENTS.SUCCESS });
         return response;
       })
       .catch(err => {
-        this.handleResponse({ error: err, event: 'ERROR' });
+        this.handleResponse({ error: err, event: EVENTS.ERROR });
         if (!isErrorSilent) {
           throw err;
         }
       });
   };
 
-  handleResponse = ({ error, event, response }: { error?: any, event: string, response?: any }) => { // eslint-disable-line
+  handleResponse = ({ error, event, response }: { error?: any, event: EVENTS.SUCCESS | EVENTS.ERROR, response?: any }) => { // eslint-disable-line
     const { setResponse } = this.props;
     if (!this._mounted) {
       return;
     }
+    const value = {
+      ...(event === EVENTS.SUCCESS ? { response } : {}),
+      ...(event === EVENTS.ERROR ? { error } : {})
```

Comment:
Correct me if I'm wrong but won't updating state like this lead to cases where both error and response are populated?

Reply:
Unless `event` is both `EVENTS.SUCCESS` and `EVENTS.ERROR` then it will not.